tests/components/landisgyr_heat_meter/test_sensor.py

```diff
@@ -199,3 +210,69 @@ async def test_restore_state(mock_heat_meter, hass: HomeAssistant) -> None:
     assert state
     assert state.state == "devicenr_789"
     assert state.attributes.get(ATTR_STATE_CLASS) is None
+
+
+@patch(API_HEAT_METER_SERVICE)
+async def test_exception_on_polling(mock_heat_meter, hass: HomeAssistant) -> None:
+    """Test sensor."""
+    entry_data = {
+        "device": "/dev/USB0",
+        "model": "LUGCUH50",
+        "device_number": "123456789",
+    }
+    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id=DOMAIN, data=entry_data)
+    mock_entry.add_to_hass(hass)
+
+    # First setup normally
+    mock_heat_meter_response = MockHeatMeterResponse(
+        heat_usage_gj=123,
+        volume_usage_m3=456,
+        heat_previous_year_gj=111,
+        device_number="devicenr_789",
+        meter_date_time=dt_util.as_utc(datetime.datetime(2022, 5, 19, 19, 41, 17)),
+    )
+
+    mock_heat_meter().read.return_value = mock_heat_meter_response
+
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await async_setup_component(hass, HA_DOMAIN, {})
+    await hass.async_block_till_done()
+    await hass.services.async_call(
+        HA_DOMAIN,
+        SERVICE_UPDATE_ENTITY,
+        {ATTR_ENTITY_ID: "sensor.heat_meter_heat_usage"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    # check if initial setup succeeded
+    assert len(hass.states.async_all()) == 27
+    state = hass.states.get("sensor.heat_meter_heat_usage")
+    assert state
+    assert state.state == "34.16669"
+
+    # Now 'disable' the connection and wait for polling
+    mock_heat_meter.reset_mock()
+    mock_heat_meter().read.side_effect = serial.serialutil.SerialException
+    async_fire_time_changed(hass, dt_util.utcnow() + POLLING_INTERVAL)
+    await hass.async_block_till_done()
+    mock_heat_meter.assert_called_once()
```

Comment:
What I'm trying to do is see that it handles the serial connection being away temporarilly. Since it's a cable that can be unplugged. This will cover the code in coordinator.py on line 36, 37.

Reply:
Now it's me getting confused. It sounds right, but if I try it, the test fails with "_AssertionError: Expected 'read' to have been called once. Called 0 times._"

Reply:
As I've said... very confusing.
I'll mark it back to draft until it's all figured out...

Reply:
I've adjusted the test to focus on the code being tested. I it better understandable now?